common/src/main/java/org/jboss/hal/testsuite/cli/DomainManager.java

```diff
@@ -472,4 +475,105 @@ public boolean waitUntilAvailable(String host, long timeout) {
         }
         return isRunning;
     }
+
+    @Override
+    public void reloadIfRequiredAndWaitUntilRunning(long timeout) {
```

Comment:
I suggest changing name, in order to not be confusing with general `waitUntilAvailable` which checks only Host.